pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -450,12 +451,14 @@ func (lbaas *LbaasV2) createLoadBalancer(service *corev1.Service, name, clusterN
 
 	if vipPort != "" {
 		createOpts.VipPortID = vipPort
+	} else if lbClass != nil && lbClass.NetworkID != "" {
+		createOpts.VipNetworkID = lbClass.NetworkID
+	} else if lbaas.opts.NetworkID != "" {
+		createOpts.VipNetworkID = lbaas.opts.NetworkID
+	} else if lbClass != nil && lbClass.SubnetID != "" {
+		createOpts.VipSubnetID = lbClass.SubnetID
 	} else {
-		if lbClass != nil && lbClass.SubnetID != "" {
-			createOpts.VipSubnetID = lbClass.SubnetID
-		} else {
-			createOpts.VipSubnetID = lbaas.opts.SubnetID
-		}
+		createOpts.VipSubnetID = lbaas.opts.SubnetID
 	}
 
```

Comment:
Logic is not right here, the network_id and subnet_id can both be provided.

Reply:
fixed! can you review. Thanks